src/Agent/NewRelic/Agent/Core/DataTransport/ConnectionManager.cs

```diff
@@ -35,19 +36,20 @@ public class ConnectionManager : ConfigurationBasedService, IConnectionManager
         private int _connectionAttempt = 0;
         private bool _started;
         private readonly object _syncObject = new object();
+        private bool _runtimeConfigurationUpdated = false;
```

Comment:
After reviewing the expected usage patterns of this field, I believe read/write access to it should be wrapped in a lock. There is a low possibility that a read and write may occur at the exact same time. The logic in the rest of the PR is solid, but we want to avoid any 1:1mil crashes due to read/write conflicts.

Reply:
I tried a few combinations of locking around the bool, but the method in the PR was the one that worked.